content/blog/2025-05-13-brilooped/index.md

```diff
@@ -0,0 +1,443 @@
+- The condition is evaluated before each iteration. If true, the body executes; if false, the loop terminates.
+- The instructions within the loop body are responsible for potentially updating the condition variable.
+
+Example usage:
+
+```json
+{
+  "args": [
+    "cond"
+  ],
+  "children": [
+    [
+      {
+        "args": [
+          "a",
+          "b"
+        ],
+        "dest": "temp",
+        "op": "add",
+        "type": "int"
+      },
+      {
+        "args": ["b", "ten"],
+        "dest": "cond",
+        "op": "le",
+        "type": "bool"
+      }
+    ]
+  ],
+  "op": "while"
+},
+```
+
+```
+ten: int = const 10;
+a: int = const 1;
+b: int = const 0;
+cond: bool = id true;
+while cond {
+    b: int = add a b;
+    cond: bool = le b ten;
+}
+```
+
+### `if`
+
+The `if` operation implements conditional execution:
+
+- It takes exactly one argument: a string containing the name of a boolean variable that serves as the condition.
+- It includes a `children` field with an array of arrays of `Instr` objects.
+  - This will contain 1 or 2 arrays, belonging to the True and False blocks respectively.
+  - When the condition is true, the first array of `Instrs` objects in the `children` is executed.
+  - When it is false, the second array is executed.
+
+Example usage:
+
+```json
+{
+  "args": ["cond"],
+  "children": [
+    [
+      {
+        "args": ["a", "b"],
+        "dest": "temp",
+        "op": "add",
+        "type": "int"
+      }
+    ],
+    [
+      {
+        "args": ["a", "b"],
+        "dest": "temp",
+        "op": "sub",
+        "type": "int"
+      }
+    ]
+  ],
+  "op": "if"
+}
+```
+
+```
+cond: bool = id true;
+if cond
+then {
+    temp: int = add a b;
+}
+else {
+    temp: int = sub a b;
+}
+
+```
+
+### The `break` and `continue` Operations
+
+Two additional operations provide finer control within loops:
+
+- `break`: Takes one constant argument and exits `value` loops. This brings execution to the top of this loop.
+- `continue`: Takes one constant argument and exits `value` loops. This continues execution at the following instruction.
+
+For example the following program will print 1 and 2 forever:
+
+```
+cond: bool = const true;
+one: int = const 1;
+two: int = const 2;
+while cond {
+    print one;
+    while cond {
+        print two;
+        continue 1;
+    }
+}
+```
+
+Whereas this program will print 1 followed by infinite twos:
+
+```
+cond: bool = const true;
+one: int = const 1;
+two: int = const 2;
+while cond {
+    print one;
+    while cond {
+        print two;
+        continue 0;
+    }
+}
+```
+
+## Interpreter Changes
+
+### New Language Constructs
+
+We expanded Bril's control flow capabilities by introducing five new constructs to the interpreter type system:
+
+- `while` - For creating loops
+- `block` - For grouping instructions
+- `if` - For conditional execution
+- `break` - For exiting loops
+- `continue` - For skipping to the next loop iteration
```

Comment:
I think you've already covered this above? As in, you've already explained the set of new operations you've added and what they mean, so you don't need this section here. You can just say, in one sentence, that you implemented each of the new constructs in the reference interpreter.